src/lib.rs

```diff
@@ -148,6 +148,15 @@
 #![deny(missing_docs)]
 #![deny(missing_debug_implementations)]
 
+// Allow clippy warnings when we aren't building with clippy.
+#![allow(unknown_lints)]
+// False positives with `fallible_iterator`.
+#![allow(should_implement_trait)]
+// Many false positives involving `continue`.
+#![allow(never_loop)]
+// False positives when block expressions are used inside an assertion.
+#![allow(panic_params)]
```

Comment:
These are a bit unfortunate. Are there upstream bugs filed against clippy for the worst offenders here that we can link to?